tarantool_test.go

```diff
@@ -3150,6 +3150,91 @@ func TestConnectionFeatureOptsImmutable(t *testing.T) {
 	require.True(t, connected, "Reconnect success")
 }
 
+func TestErrorExtendedInfoBasic(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("not a Lua code", []interface{}{})
+	require.NotNilf(t, err, "expected error on invalid Lua code")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "LuajitError",
+		File:  "eval",
+		Line:  uint32(1),
+		Msg:   "eval:1: unexpected symbol near 'not'",
+		Errno: uint32(0),
+		Code:  uint32(32),
+	}
+
+	test_helpers.EqualBoxErrors(t, expected, *ttErr.ExtendedInfo)
+}
+
+func TestErrorExtendedInfoStack(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("error(chained_error)", []interface{}{})
+	require.NotNilf(t, err, "expected error on explicit error raise")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "ClientError",
+		File:  "config.lua",
+		Line:  uint32(214),
+		Msg:   "Timeout exceeded",
+		Errno: uint32(0),
+		Code:  uint32(78),
+		Prev: &BoxError{
+			Type:  "ClientError",
+			File:  "config.lua",
+			Line:  uint32(213),
+			Msg:   "Unknown error",
+			Errno: uint32(0),
+			Code:  uint32(0),
+		},
+	}
+
+	test_helpers.EqualBoxErrors(t, expected, *ttErr.ExtendedInfo)
+}
+
+func TestErrorExtendedInfoFields(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("error(access_denied_error)", []interface{}{})
+	require.NotNilf(t, err, "expected error on forbidden action")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "AccessDeniedError",
+		File:  "/__w/sdk/sdk/tarantool-2.10/tarantool/src/box/func.c",
+		Line:  uint32(535),
```

Comment:
The line check will fail on new Tarantool versions.

Reply:
Yeah, so test error comparison ignores them